test/integration/data-types.test.js

```diff
@@ -248,6 +248,43 @@ describe(Support.getTestDialectTeaser('DataTypes'), () => {
     });
   });
 
+  if (dialect === 'mysql') {
```

Comment:
I think we could make this a simple unit test, not mysql specific?

Reply:
This behavior is specific to MySQL. On other platforms, where `TRUE` and `FALSE` are not aliases for `1` and `0` respectively, passing a boolean literal where a number is expected will result in an error.